Retrieve school experience events

peteryates opened the pull request:
In an attempt to treat School Experience as another of GIT's transactions we're going to pull their transaction data into BQ with ours.

This is quite tricky as they're held in a tabular data structure in a single text column (`dfe_notesforclassroomexperience`). As a result, we need to do a couple of rounds of manipulation via [common table expressions](https://docs.microsoft.com/en-us/sql/t-sql/queries/with-common-table-expression-transact-sql?view=sql-server-ver15).

Unfortunately the data within isn't all perfectly-aligned either, so we need to do a bit of work to pull it out in a sensible fashion.

peteryates committed:
Retrieve school experience events

This is quite tricky as they're held in a textual data structure in
a single text column. Unfortunately the data within isn't all
perfectly-aligned either, so we need to do a bit of work to pull
it out in a sensible fashion.

peteryates commented on views/school_experience.sql:
```diff
@@ -0,0 +1,139 @@
+
+        where
+            dfe_notesforclassroomexperience is not null  -- only target SE-related rows
+            and value <> 'RTBF'                          -- no idea what RTBF stands for but looks like 'no data'!
+            and value <> 'none'                          -- no data (why are these here?)
+            and value not like 'RECORDED%'               -- not the header row
+            and len(trim(value)) > 1                     -- not the empty row beneath the header
```
These criteria are probably superfluous now the date 'plucking' is in, but they at least provide a bit of context on the other stuff that's crammed in this column so I left them in.

peteryates committed:
Add table overview to fields doc

MylesJarvis commented on views/school_experience.sql:
```diff
@@ -0,0 +1,140 @@
+
+        where
+            dfe_notesforclassroomexperience is not null  -- only target SE-related rows
+            and value <> 'RTBF'                          -- no idea what RTBF stands for but looks like 'no data'!
```
Right to be forgotten

peteryates committed:
Split on the newline rather than the return

peteryates committed:
Add explanation of RTBF

These records are where people have requested that their data is removed
from the system.
